ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/attestation/ValidateableAttestation.java

```diff
@@ -50,15 +57,38 @@ private ValidateableAttestation(
     this.hashTreeRoot = Suppliers.memoize(attestation::hash_tree_root);
   }
 
+  public Optional<IndexedAttestation> getMaybeIndexedAttestation() {
+    return maybeIndexedAttestation;
+  }
+
   public IndexedAttestation getIndexedAttestation() {
     return maybeIndexedAttestation.orElseThrow(
         () ->
             new UnsupportedOperationException(
                 "ValidateableAttestation does not have an IndexedAttestation yet."));
   }
 
-  public void setIndexedAttestation(IndexedAttestation maybeIndexedAttestation) {
-    this.maybeIndexedAttestation = Optional.of(maybeIndexedAttestation);
+  public Optional<Bytes32> getMaybeRandaoMix() {
```

Comment:
I'm not really a fan of the `Maybe` in these method names.  I'd be tempted to just move to having a single `getRandaoMix` which returns an optional and do the `orElseThrow` in the caller if desired (and same for `getIndexedAttestation`).

Reply:
Done.